### PR TITLE
refactor: define radix context for reusability as a React provider

### LINKED ISSUE #19: Tech Debt: define radix context for reusability as a React provider
**Describe the tech debt**
We need reusable Radix context and hook. Please utilise the Radix [sample implementation](https://github.dev/radixdlt/gumball-club/blob/main/dapp/src/app/radix/radix-context.ts).

**Which areas / pages are affected**
Main page and Radix login button theme.

**Expected behaviour**
App should be able to login user with persona.

**Additional context**
Try to solve the theme change issue and the Radix Wallet Connect should use the selected light/dark theme.

### PR DESCRIPTION
**Pull request by attilacsanyi**
Fix #19

### CODE REVIEW

File: src/app/provider/radix-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      {children}
+    </RadixProviderContext.Provider>
+  );
+};
+
+export const useDappToolkit = () => {
+  const context = useContext(RadixProviderContext);
+
+  if (context === undefined)
+    throw new Error('useDappToolkit must be used within a RadixProvider');
```
Comment (attilacsanyi) [inline src/app/provider/radix-provider.tsx:35]: Can you help me how to trigger these error case? I removed the wrapper provider, but not triggered 🙇

Comment (AhsenRiaz) [reply src/app/provider/radix-provider.tsx:35]: I tried `context === null` and the error was triggered.

Comment (attilacsanyi) [reply src/app/provider/radix-provider.tsx:35]: Yeah that is the trivial case, however **in React you need to wrap your JSX with the proper provider**, so if you miss that then this check is for that scenario, but even if I not provide the provider I cannot get this error which should be the case. Do you know what I mean?

Comment (attilacsanyi) [reply src/app/provider/radix-provider.tsx:35]: I found the issue, check my latest relevant [commit](52e9163fd35b3157b156c16e579ff736401d603a).

**Review by AhsenRiaz — APPROVED** [review]

**Pull request merged**